src/clientHandler.ts

```diff
@@ -108,19 +108,64 @@ export class ClientHandler {
   private async getServerOptions(): Promise<ServerOptions> {
     const cmd = await this.lsPath.resolvedPathToBinary();
     const serverArgs = config('terraform').get<string[]>('languageServer.args', []);
+    const procEnv = this.getProcessEnvironment();
+
     const executable: Executable = {
       command: cmd,
       args: serverArgs,
-      options: {},
+      options: {
+        env: procEnv,
+      },
     };
+
     const serverOptions: ServerOptions = {
       run: executable,
       debug: executable,
     };
+
     this.outputChannel.appendLine(`Launching language server: ${cmd} ${serverArgs.join(' ')}`);
     return serverOptions;
   }
 
+  private getProcessEnvironment() {
+    // deep object clone without the bother of custom traversal
+    const procEnv = JSON.parse(JSON.stringify(process.env));
```

Comment:
This looks a bit awkward to me, but I'm not really familiar with data copying practices in JavaScript myself, so I'd defer to you and @dbanck here.

Reply:
It's a useful way to deep clone an object in JavaScript, but it won't clone any function or symbol properties (which isn't necessary here). We might even consider using `Object.assign` for shallow cloning here because `process.env` should just contain strings and not any objects/arrays.

Reply:
I've used `Object.assign` and prototype builtins for shallow cloning objects in the past with varying success. TS strict does not like the `any` usage for prototype proprty discovery and doing a search on `Object.assign` finds [several kinds of problems](https://stackoverflow.com/questions/34504682/js-does-object-assign-create-deep-copy-or-shallow-copy) with using it and shadow copies.

For me, I chose `JSON.parse` largely from subjective rather than objective reasons. I'd rather use a built-in method here for this than custom code I wrote, or `Object.assign` which changes behavior based on what's stored in the object and what node version we run.

Reply:
I see, that's fair! I suppose it looks easy to read at least - I have seen some other snippets of code copying data in JS which look far more complicated! 😅

Reply:
My only worry was perhaps the performance hit with the string conversions, but I suppose most of the time there won't be as many variables to make this a big deal.

Reply:
> TS strict does not like the any usage for prototype property discovery

I don't understand what you are referring to, can you explain?

---

Thanks for the hints about `Object.assign` I'm fine with avoiding it. What do you think of using `{ ...process.env }` for creating a shallow copy?

Reply:
There is no performance difference between using the `JSON` method and `{ ... process.env }` so I changed it to `{ ... proces.env } as the more important bit here is trying to solve the 'can't find terraform' part and not how we access environment variabls